core/Symbols.cc

```diff
@@ -725,7 +725,12 @@ string ArgInfo::argumentName(const GlobalState &gs) const {
     } else {
         // positional arg
         if (loc.exists()) {
-            return loc.source(gs);
+            auto src = loc.source(gs);
+            if (flags.isBlock && src.size() > 0 && src[0] == '&') {
```

Comment:
This seems like a hack, can we convert this to an ENFORCE on creation instead and fix early phases to not pass such `loc`s for blocks?

Reply:
ArgInfo is a builder right now (you directly twiddle the bits). I don't really want to convert it to a constructor just to add this ENFORCE, as I think that after changing this to be done in the parser it will likely never break.

Also it turns out that doing this in parser/Builder.cc was **way** easier than I anticipated it would be (I'd never had to make a change to the parser so I figured it'd be really annoying, but it turns out to be fewer lines of code changed than my original implementation.)